src/typed-geometry/functions/matrix/perspective.hh

```diff
@@ -7,6 +7,15 @@
 #include <typed-geometry/feature/assert.hh>
 #include <typed-geometry/functions/basic/scalar_math.hh>
 
+namespace
```

Comment:
global anon namespace in headers leak everywhere. you probably want a tg::detail namespace